module2/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,53 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+#include <utility>
+
+compressedImage compressGrayscale(decompressedImage& input) {
+    compressedImage result;
+    result.reserve(width * height);
+
+    std::for_each(begin(input), end(input), [&result](decompressedLine& line){
+        compressedImage compressedLine = compressSingleLine(line);
+        std::transform(begin(compressedLine), end(compressedLine), std::back_inserter(result), [](const auto& pair){
+            return std::make_pair(pair.first, pair.second);
+        });
+    });
+
+    result.shrink_to_fit();
+    return result;
+}
+
+compressedImage compressSingleLine(decompressedLine& line) {
+    compressedImage buffer;
+    buffer.reserve(width);
+    std::pair<uint8_t, uint8_t> current = {line.front(), 0};
```

Comment:
Maybe you should make alias for std::pair<uint8_t, uint8_t>. You use it more than once.

Reply:
+1